### PR TITLE
Log request headers

### PR DESCRIPTION
**Pull request by amit-mb**
Logs requests on a per tab basis. Also fixes the button states across tabs.

### CODE REVIEW

File: app/manifest.json [hunk 1]
```diff
@@ -5,13 +5,12 @@
   "version": "0.1",
 
   "background": {
-    "scripts": ["background.js"]
+    "scripts": ["background.js"],
+    "persistent": true
```
Comment (revathskumar) [inline app/manifest.json]: why we need this?

File: app/background.ts [hunk 2]
```diff
@@ -8,83 +8,84 @@ interface Data {
...
-  findItem = (tabId : number) => {
-    // Not using a polyfill because this is supported in Chrome v. > 45. We
-    // might have to add some restriction in manifest related to this
-    const data = this.data.find(item => item.tabId === tabId) || DEFAULT_DATA;
-
-    // Lazy code for object cloning
-    return JSON.parse(JSON.stringify(data));
+class BackgroundWorker {
+  currentTab:number = -1;
+  data = {}
```
Comment (revathskumar) [inline app/background.ts]: don't we need to specify type here?

File: app/message_service.ts [hunk 3]
```diff
@@ -18,17 +18,12 @@ export function resetData(tabId: number) {
   chrome.tabs.sendMessage(tabId, {message: "RESET_DATA"});
 }
 
-export function getEnabledStatus(callback: GenericCallback) {
-  chrome.runtime.sendMessage({message: "GET_ENABLED_STATUS"}, callback);
+export function getEnabledStatus(tabId:number, callback: GenericCallback) {
+  chrome.runtime.sendMessage({message: "GET_ENABLED_STATUS", tabId: tabId }, callback);
```
Comment (revathskumar) [inline app/message_service.ts]: instead of `{ tabId: tabId }` we can use `{ tabId }`

File: app/message_service.ts [hunk 4]
```diff
@@ -18,17 +18,12 @@ export function resetData(tabId: number) {
...
 }
 
-export function getRequests(tabId, callback: GenericCallback) {
-  chrome.runtime.sendMessage(chrome.runtime.id, {message: "GET_REQUESTS", tabId: tabId }, {}, callback);
+export function getRequests(tabId:number, callback: GenericCallback) {
+  chrome.runtime.sendMessage({message: "GET_REQUESTS", tabId: tabId }, {}, callback);
```
Comment (revathskumar) [inline app/message_service.ts]: instead of `{ tabId: tabId }` we can use `{ tabId }`

File: app/manifest.json (hunk 1 above)
Comment (amit-mb) [reply app/manifest.json]: ```"persistent": true``` is by default. We can revert that back

**Pull request merged**